x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/EnrollmentToken.java

```diff
@@ -0,0 +1,119 @@
+
+import static org.elasticsearch.common.xcontent.ConstructingObjectParser.constructorArg;
+
+public class EnrollmentToken {
+    private final String apiKey;
+    private final String fingerprint;
+    private final String version;
+    private final List<String > boundAddress;
+
+    public String getApiKey() { return apiKey; }
+    public String getFingerprint() { return fingerprint; }
+    public String getVersion() { return version; }
+    public List<String> getBoundAddress() { return boundAddress; }
+
+    private static final ParseField API_KEY = new ParseField("key");
+    private static final ParseField FINGERPRINT = new ParseField("fgr");
+    private static final ParseField VERSION = new ParseField("ver");
+    private static final ParseField ADDRESS = new ParseField("adr");
+
+    @SuppressWarnings("unchecked")
+    public static final ConstructingObjectParser<EnrollmentToken, Void> PARSER = new ConstructingObjectParser<>("enrollment_token", false,
+        a -> new EnrollmentToken((String) a[0], (String) a[1], (String) a[2], (List<String>) a[3]));
+
+    static {
+        PARSER.declareString(constructorArg(), API_KEY);
+        PARSER.declareString(constructorArg(), FINGERPRINT);
+        PARSER.declareString(constructorArg(), VERSION);
+        PARSER.declareStringArray(constructorArg(), ADDRESS);
+    }
+    /**
+     * Create an EnrollmentToken
+     *
+     * @param apiKey         API Key credential in the form apiKeyId:ApiKeySecret to be used for enroll calls
+     * @param fingerprint    hex encoded SHA256 fingerprint of the HTTP CA cert
+     * @param version        node version number
+     * @param boundAddress   IP Addresses and port numbers for the interfaces where the Elasticsearch node is listening on
+     */
+    public EnrollmentToken(String apiKey, String fingerprint, String version, List<String> boundAddress) {
+        this.apiKey = Objects.requireNonNull(apiKey);
+        this.fingerprint = Objects.requireNonNull(fingerprint);
+        this.version = Objects.requireNonNull(version);
+        this.boundAddress = Objects.requireNonNull(boundAddress);
+    }
+
+    public String getRaw() throws Exception {
+        final XContentBuilder builder = JsonXContent.contentBuilder();
+        builder.startObject();
+        builder.field("ver", version);
+        builder.startArray("adr");
+        for (String bound_address : boundAddress) {
+            builder.value(bound_address);
+        }
+        builder.endArray();
+        builder.field("fgr", fingerprint);
+        builder.field("key", apiKey);
+        builder.endObject();
+        return Strings.toString(builder);
+    }
+
+    public String getEncoded() throws Exception {
+        final String jsonString = getRaw();
+        return Base64.getUrlEncoder().encodeToString(jsonString.getBytes(StandardCharsets.UTF_8));
+    }
+
+    /**
+     * Decodes and parses an enrollment token from it's serialized form (created with {@link EnrollmentToken#getEncoded()}
```

Comment:
```suggestion
     * Decodes and parses an enrollment token from its serialized form (created with {@link EnrollmentToken#getEncoded()}
```

Reply:
If I had a euro for every time I've made that mistake...